### PR TITLE
Fixes an issue with set and table support in syslwrapper

### PR DESCRIPTION
**Pull request by cuminandpaprika**
Changes proposed:

- Add support for set in responses and type definitions
- Add basic support for relation type in sysl type definitions

Checklist:
- [x] Added related tests
- [ ] Made corresponding changes to the documentation

### CODE REVIEW

File: pkg/syslwrapper/app.go [hunk 1]
```diff
@@ -96,7 +96,9 @@ func (am *AppMapper) IndexTypes() map[string]*sysl.Type {
 func (am *AppMapper) ConvertTypes() map[string]*Type {
 	simpleTypes := make(map[string]*Type)
 	for typeName, syslType := range am.Types {
-		simpleTypes[typeName] = am.MapType(syslType)
+		if am.MapType(syslType) != nil {
+			simpleTypes[typeName] = am.MapType(syslType)
```
Comment (ashwinsajiv) [inline pkg/syslwrapper/app.go:100]: Can this -am.MapType(syslType) be stored in a meaningful name? So can be made like -
if meaningfulName := am.MapType(syslType); meaningfulName != nil {
        simpleTypes[typeName] = meaningfulName
}

File: pkg/syslwrapper/app.go [hunk 2]
```diff
@@ -121,7 +123,9 @@ func (am *AppMapper) mapAttributes(attributes map[string]*sysl.Attribute) map[st
 func (am *AppMapper) mapTypes(appName string, syslTypes map[string]*sysl.Type) map[string]*Type {
 	simpleTypes := make(map[string]*Type, 15)
 	for typeName := range syslTypes {
-		simpleTypes[typeName] = am.MapType(am.Types[appName+":"+typeName])
+		if am.MapType(am.Types[appName+":"+typeName]) != nil {
+			simpleTypes[typeName] = am.MapType(am.Types[appName+":"+typeName])
```
Comment (ashwinsajiv) [inline pkg/syslwrapper/app.go:127]: Similar changes to this

File: pkg/syslwrapper/app.go (hunk 1 above)
Comment (cuminandpaprika) [reply pkg/syslwrapper/app.go:100]: Yup, agreed, can definitely do that

File: pkg/syslwrapper/app.go (hunk 2 above)
Comment (cuminandpaprika) [reply pkg/syslwrapper/app.go:127]: No worries, will do!

Comment (cuminandpaprika) [reply pkg/syslwrapper/app.go]: Fixed up in https://github.com/anz-bank/sysl/pull/831/commits/49670ff4551670504b4e47b555a6e4b429979239

File: pkg/syslwrapper/app.go (hunk 1 above)
Comment (cuminandpaprika) [reply pkg/syslwrapper/app.go]: Fixed up in https://github.com/anz-bank/sysl/pull/831/commits/49670ff4551670504b4e47b555a6e4b429979239

**Pull request merged**